utils/src/integ-test/java/uk/gov/moj/sdt/test/utils/DBUnitUtility.java

```diff
@@ -1,635 +1,620 @@
+
+            // Get the connection already setup for DBUNIT.
+            final IDatabaseConnection dbConnection = getConnectionInstance(schemaName);
+            final Connection connection = dbConnection.getConnection();
+
+            // Call the reset stored procedure which should have been loaded into the database by ANT script.
+            final CallableStatement cs = connection.prepareCall("{call SDT_OWNER.Finish_Dbunit_Load(?)}");
+            cs.setString("p_SchemaName", schemaName);
+
+            // Close result set and calleable statement.
+            final ResultSet rs = cs.executeQuery();
+            if (rs != null) {
+                rs.close();
+            }
+            cs.close();
+        } catch (final SQLException e) {
+            LOGGER.error(e.toString());
+        }
+    }
+
+    /**
+     * Clean database.
+     *
+     * @param targetSchema  the target schema
+     * @param reloadRefdata the reload refdata
+     */
+    public static void cleanDatabase(final String targetSchema, final boolean reloadRefdata) {
+        try {
+            // database connection
+            LOGGER.debug("Loading data for schema " + targetSchema);
+            final IDatabaseConnection dbConnection = getConnectionInstance(targetSchema);
+
+            // Create a DBUnit Dataset builder using the target schemas DTD
+            final FlatXmlDataSetBuilder builder = new FlatXmlDataSetBuilder();
+            builder.setDtdMetadata(true);
+            builder.setMetaDataSetFromDtd(DBUnitUtility.class.getResourceAsStream("/DTD/" + targetSchema + ".dtd"));
+            builder.setColumnSensing(true);
+
+            // Generate the clean dataset, if loadRefData is true use
+            // clean_all.xml otherwise use clean.xml
+            final String cleanDataPath = "/clean/" + targetSchema + "_clean" + (reloadRefdata ? "_all" : "") + ".xml";
+            final IDataSet cleanDataset = builder.build(DBUnitUtility.class.getResourceAsStream(cleanDataPath));
+
+            // Disable triggers & constraints
+//            prepareForDbunitLoad(targetSchema);
+
+            // Clean the DB
+            LOGGER.debug("Cleaning database");
+            DatabaseOperation.TRUNCATE_TABLE.execute(dbConnection, cleanDataset);
+            dbConnection.getConnection().commit();
+
+            // If there's reference data insert it
+            // InputStream refDataStream =
+            // DBUnitUtility.class.getResourceAsStream("/refdata/" +
+            // targetSchema +
+            // "_refdata.xml");
+            // if (refDataStream != null && reloadRefdata)
+            // {
+            // LOGGER.debug("Inserting reference data");
+            // IDataSet referenceDataset = builder.build(refDataStream);
+            // DatabaseOperation.INSERT.execute(connection, referenceDataset);
+            // connection.getConnection().commit();
+            // }
+
+            // Enable triggers & constraints
+//            finishDbunitLoad(targetSchema);
```

Comment:
Don't forget to remove this if it is no longer needed.

Reply:
keepiing these for now, uncommented it now.
will delete DBUnitUtility.java, once we are happy with the changes done and having working application